diff --git a/setup.py b/setup.py
index 982433c..c87baad 100644
--- a/setup.py
+++ b/setup.py
@@ -85,7 +85,7 @@ setup(
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
     ],
-    packages=find_packages(exclude=["ez_setup"]),
+    packages=find_packages(exclude=["ez_setup","tests","tests.*"]),
     install_requires=install_requires,
     python_requires=">= 3.7",
     include_package_data=True,
