diff --git CMakeLists.txt CMakeLists.txt
index cf7a2d55b..a844fc93b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,16 @@ include(Version)
 set(CMAKE_ADDITIONAL_DEPS_PATH "${CMAKE_CURRENT_BINARY_DIR}/deps" CACHE PATH "Additional directory to search for libraries and headers")
 list(APPEND CMAKE_PREFIX_PATH ${CMAKE_ADDITIONAL_DEPS_PATH})
 
+
+############################################################################
+# User selectable paths
+############################################################################
+set(INSTALL_BINDIR "bin/" CACHE PATH "Binary install destination")
+set(INSTALL_GAMES_BINDIR "games/bin" CACHE PATH "Games binary install destination")
+set(INSTALL_DATAROOTDIR "share/" CACHE PATH "Data root install destination")
+set(INSTALL_DATADIR "${INSTALL_DATAROOTDIR}" CACHE PATH "Data install destination")
+
+
 ############################################################################
 # User selectable options
 ############################################################################
@@ -167,7 +177,13 @@ if(WIN32 AND MINGW)
 endif()
 
 if(UNIX)
-	add_definitions("-DOC_SYSTEM_DATA_DIR=\"${CMAKE_INSTALL_PREFIX}/share/games/openclonk\"")
+	if(NOT IS_ABSOLUTE "${INSTALL_DATADIR}")
+		set(ABSOLUTE_INSTALL_DATADIR "${CMAKE_INSTALL_PREFIX}/${INSTALL_DATADIR}")
+	else()
+		set(ABSOLUTE_INSTALL_DATADIR "${INSTALL_DATADIR}")
+	endif()
+
+	add_definitions("-DOC_SYSTEM_DATA_DIR=\"${ABSOLUTE_INSTALL_DATADIR}/openclonk\"")
 endif()
 
 if(APPLE)
@@ -1505,7 +1521,7 @@ foreach(group ${OC_C4GROUPS})
 			DEPENDS "${native_c4group}"
 			VERBATIM
 		)
-		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${group} DESTINATION share/games/openclonk)
+		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${group} DESTINATION "${ABSOLUTE_INSTALL_DATADIR}/openclonk")
 	endif()
 endforeach()
 
@@ -1515,12 +1531,12 @@ if(NOT APPLE)
 	set_property(TARGET groups PROPERTY FOLDER "Setup")
 	add_dependencies(data groups)
 	# Install binaries
-	install(TARGETS c4group DESTINATION bin)
+	install(TARGETS c4group DESTINATION "${INSTALL_BINDIR}")
 endif()
 if(NOT HEADLESS_ONLY)
 	if(NOT APPLE)
 		# Install new files
-		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.desktop DESTINATION share/applications)
+		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.desktop DESTINATION "${INSTALL_DATAROOTDIR}/applications")
 		# Update the MIME cache -- this makes the clonk:// protocol handler actually work
 		install(CODE "execute_process(COMMAND update-desktop-database)")
 
@@ -1528,7 +1544,7 @@ if(NOT HEADLESS_ONLY)
 		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.appdata.xml DESTINATION share/appdata)
 
 		# Install binaries
-		install(TARGETS openclonk DESTINATION games)
+		install(TARGETS openclonk DESTINATION "${INSTALL_GAMES_BINDIR}")
 	else()
 		install(TARGETS openclonk
 			BUNDLE DESTINATION .
