--- Variables.mk	2009-03-30 23:01:02.000000000 +0200
+++ Variables.mk.modified	2009-03-30 23:06:54.000000000 +0200
@@ -40,7 +40,7 @@
 
 ############################################## INCLUDES
 
-CXXFLAGS_Linux = -I$(EXT)/inst/include/libxml2
+CXXFLAGS_Linux = -I /usr/include/libxml2 -I/usr/include/fmodex
 CXXFLAGS_Darwin = -isysroot ${SDK} -mmacosx-version-min=10.4 -I${SDK}/usr/include/libxml2 -I"/Developer/FMOD Programmers API/api/inc"
 CXXFLAGS = $(CXXFLAGS_$(OS)) -I$(EXT)/inst/include -I$(HOME)/cvs/code/nel/include -I$(HOME)/cvs/code/nel/src
 
