--- ac/source/src/Makefile	2009-03-28 13:19:34.000000000 +0100
+++ makefile-modified/Makefile	2009-03-28 13:18:18.000000000 +0100
@@ -6,7 +6,7 @@
 CXXFLAGSSERVER=$(CXXOPTFLAGS) $(INCLUDES1) $(CUSTOMFLAGS)
 
 PLATFORM_PREFIX=native
-CLIENT_LIBS=-L/usr/X11R6/lib -L../enet -lenet `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -ljpeg
+CLIENT_LIBS=-L/usr/X11R6/lib -L../enet ../enet/libenet.a  `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -ljpeg
 CLIENT_OBJS= \
 	audiomanager.o \
 	client.o \
