diff --git a/bear-factory/CMakeLists.txt b/bear-factory/CMakeLists.txt
index 6c01034..ed900b7 100644
--- a/bear-factory/CMakeLists.txt
+++ b/bear-factory/CMakeLists.txt
@@ -14,7 +14,7 @@ if( NOT BEAR_FACTORY_INSTALL_LIBRARY_DIR )
       "${BEAR_FACTORY_INSTALL_EXECUTABLE_DIR}"
       )
   else( WIN32 )
-    set( BEAR_FACTORY_INSTALL_LIBRARY_DIR lib )
+    set( BEAR_FACTORY_INSTALL_LIBRARY_DIR lib64 )
   endif( WIN32 )
 endif( NOT BEAR_FACTORY_INSTALL_LIBRARY_DIR )
 
diff --git a/bear-factory/bear-editor/src/bf/code/accelerator_table.cpp b/bear-factory/bear-editor/src/bf/code/accelerator_table.cpp
index d9420eb..52fea6b 100644
--- a/bear-factory/bear-editor/src/bf/code/accelerator_table.cpp
+++ b/bear-factory/bear-editor/src/bf/code/accelerator_table.cpp
@@ -99,6 +99,6 @@ void bf::accelerator_table::on_key_pressed( wxKeyEvent& event )
   if ( it != m_accelerators.end() )
     {
       wxCommandEvent command( wxEVT_COMMAND_MENU_SELECTED, it->second );
-      m_event_handler.ProcessEvent( command );
+      //m_event_handler.ProcessEvent( command );
     }
 } // accelerator_table::on_key_pressed()
diff --git a/bear-factory/bear-editor/src/bf/code/base_editor_application.cpp b/bear-factory/bear-editor/src/bf/code/base_editor_application.cpp
index e49f15f..aeaec85 100644
--- a/bear-factory/bear-editor/src/bf/code/base_editor_application.cpp
+++ b/bear-factory/bear-editor/src/bf/code/base_editor_application.cpp
@@ -353,6 +353,6 @@ void bf::base_editor_application::remove_options_at( int index, int count )
   for ( int i=0; i != count; ++i )
     {
       --argc;
-      argv[argc] = NULL;
+     // argv[argc] = NULL;
     }
 } // base_editor_application::remove_options_at()
diff --git a/cmake-helper/bear-config.cmake b/cmake-helper/bear-config.cmake
index e66d79e..e1be421 100644
--- a/cmake-helper/bear-config.cmake
+++ b/cmake-helper/bear-config.cmake
@@ -47,7 +47,7 @@ if( NOT BEAR_ENGINE_INSTALL_LIBRARY_DIR )
       "${BEAR_ENGINE_INSTALL_EXECUTABLE_DIR}"
       )
   else( WIN32 )
-    set( BEAR_ENGINE_INSTALL_LIBRARY_DIR lib )
+    set( BEAR_ENGINE_INSTALL_LIBRARY_DIR lib64 )
   endif( WIN32 )
 endif( NOT BEAR_ENGINE_INSTALL_LIBRARY_DIR )
 
