diff -Naur src/global.cpp src-p/global.cpp
--- src/global.cpp	2007-03-07 10:34:56.000000000 +0100
+++ src-p/global.cpp	2009-08-21 15:00:08.000000000 +0200
@@ -9,6 +9,7 @@
 
 #include "Global.h"
 #include "unixutils.h"
+#include <limits.h>
 
 
 
diff -Naur src/levels.h src-p/levels.h
--- src/levels.h	2007-03-07 10:35:04.000000000 +0100
+++ src-p/levels.h	2009-08-21 15:00:34.000000000 +0200
@@ -8,7 +8,7 @@
 
 #ifndef __LEVELS_H__
 #define __LEVELS_H__
-
+#include <limits.h>
 
 class cScene
 {
