From: Julian Ospald <hasufell@gentoo.org>
Date: Mon Nov  5 17:41:16 UTC 2012
Subject: build system

	add MINIMAL_FLAGS option to preserve our system flags

--- CMakeLists.txt
+++ CMakeLists.txt
@@ -16,6 +16,9 @@
 
 set( CMAKE_BUILD_TYPE Release CACHE STRING "Build configuration (Debug, Release, RelWithDebInfo, MinSizeRel)" )
 
+# convenience option for source distros
+option( MINIMAL_FLAGS "try not to mess up system flags" 0)
+
 project( Daemon C CXX ASM )
 
 # Check if this is first run and if so set some default flags
@@ -25,11 +28,16 @@
   if( MSVC )
     set( CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /arch:SSE /fp:precise /Oy" CACHE STRING "Flags used by the compiler during release builds" FORCE )
   else()
+    if(NOT MINIMAL_FLAGS)
     set( CMAKE_C_FLAGS "-msse2 -pipe" CACHE STRING "Flags used by the compiler during all build types" FORCE )
     set( CMAKE_C_FLAGS_DEBUG "-g" CACHE STRING "Flags used by the compiler during debug builds." FORCE )
     set( CMAKE_C_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Flags used by the compiler during release minsize builds." FORCE )
     set( CMAKE_C_FLAGS_RELEASE "-DNDEBUG -O3 -ffast-math -fomit-frame-pointer -fno-strict-aliasing" CACHE STRING "Flags used by the compiler during release builds" FORCE )
     set( CMAKE_C_FLAGS_RELWITHDEBINFO "-DNDEBUG -O3 -ffast-math -fomit-frame-pointer -fno-strict-aliasing -g" CACHE STRING "Flags used by the compiler during Release with Debug Info builds." FORCE )
+    else()
+        set( CMAKE_C_FLAGS_DEBUG "" )
+        set( CMAKE_C_FLAGS_RELEASE "-DNDEBUG" )
+    endif(NOT MINIMAL_FLAGS)
   endif()
 
   set( CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "Flags used by the compiler during all build types" FORCE )
@@ -239,6 +247,7 @@
   set( CPACK_BUNDLE_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cpack/OS_X/x86_64/info.plist" )
 endif()
 
+if(NOT MINIMAL_FLAGS)
 #################
 # Compile flags #
 #################
@@ -374,6 +383,7 @@
 if( ENABLE_RELRO )
   try_linker_flag( HAVE_Z_RELRO_Z_NOW -Wl,-z,relro,-z,now )
 endif()
+endif(NOT MINIMAL_FLAGS)
 
 ######################
 # Define Build Files #
