diff --git a/CMakeLists.txt b/CMakeLists.txt
index a941087..4b6f6ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,8 +21,8 @@ set(CLAW_REVISION_NUMBER 0)
 set( CLAW_CODE_DIR claw/code/ )
 set( CLAW_INCLUDE_DIRECTORY "${CMAKE_INSTALL_PREFIX}/include/" )
 set( CLAW_INSTALLDIR_SOURCE include/claw/ )
-set( CLAW_INSTALLDIR_LIB lib/ )
-set( CLAW_INSTALLDIR_DOC share/doc/libclaw${CLAW_VERSION_MAJOR}/ )
+set( CLAW_INSTALLDIR_LIB lib64/ )
+set( CLAW_INSTALLDIR_DOC share/doc/libclaw/ )
 set( CLAW_TRANSLATIONS_INSTALL_DIR "share/locale" )
 set( CLAW_EXECUTABLE_DIR bin/ )
 
