--- /var/tmp/Directories.cpp	2007-08-02 23:32:49.000000000 +0200
+++ ./util/Directories.cpp	2007-08-02 23:34:28.000000000 +0200
@@ -54,7 +54,7 @@
 const fs::path GetGlobalDir()
 {
     if (!g_initialized) InitDirs();
-    char* dir_name = br_find_data_dir("/usr/local/share");
+    char* dir_name = br_find_data_dir("/usr/share/games");
     fs::path p(dir_name);
     std::free(dir_name);
     p /= "freeorion";
